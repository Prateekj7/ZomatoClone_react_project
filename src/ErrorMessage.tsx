import * as React from 'react';
import { Typography, makeStyles } from "@material-ui/core";

const useStyles = makeStyles({
    container: {
        color: "red",
    }
});

interface IProps{
    message: string;
}

export default function ErrorMessage(props: IProps){
    const classes = useStyles();
    return (
        <div>
            <Typography variant="h5" className={classes.container}>
                {props.message}
            </Typography>
        </div>
      );
}