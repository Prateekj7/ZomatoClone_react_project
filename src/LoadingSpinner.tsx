import * as React from 'react';
import { CircularProgress, makeStyles } from "@material-ui/core";

const useStyles = makeStyles({
    container: {
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        height: "100%",
        position: "fixed",
        left: "0px",
        top: "0px",
        width: "100%",
        ZIndexProperty : "9999"
    }
});

export default function LoadingSpinner(){
    const classes = useStyles();
    return (
        <div className={classes.container}>
            <CircularProgress />
        </div>
      );
}