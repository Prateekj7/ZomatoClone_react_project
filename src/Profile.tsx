import firebase from "firebase";
import Footer from "./Footer";
import "firebase/auth"
import { useHistory } from "react-router";
import { makeStyles, Typography, useFormControl } from "@material-ui/core";
import { useContext } from "react";
import {UserContext } from "./UserProvider"

const useStyles = makeStyles({
    container: {
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        padding: "20px",
        flexDirection: "column"
    }
});


export default function Profile(){
    const history = useHistory();
    const userContext = useContext(UserContext);
    const signOut = async () =>{
        await firebase.auth().signOut();
        history.push("/")
    }
    const styles = useStyles();
    return (
        <div className={styles.container}>
            <Typography variant="h3">Welcome to your Profile</Typography>
            <Typography variant="h5">{userContext?.displayName}</Typography>
            <Typography variant="h5">{userContext?.email}</Typography>
            <Typography variant="h5">{userContext?.uid}</Typography>
            <Footer 
                primary={{
                    text: "Sign out",
                    onClick: signOut
                }}
                secondary={{
                    text: "Home",
                    onClick: () => history.push("/")
                }}
                tertiary={{
                    text: "Back",
                    onClick: () => history.goBack()
                }}
            />
        </div>
        );
}