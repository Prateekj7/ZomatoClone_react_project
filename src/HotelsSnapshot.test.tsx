import { render } from "@testing-library/react"
import * as react from "react";
import ErrorMessage from "./ErrorMessage"
import LoadingSpinner from "./LoadingSpinner";
import Profile from "./Profile";


describe("Hotels Sanpshot test", () => {
    it("Error Message", () => {
        const {baseElement} = render(<ErrorMessage message={"test message"}/>);
        expect(baseElement).toMatchSnapshot();
    })
    it("Loading Spinner", () => {
        const {baseElement} = render(<LoadingSpinner />);
        expect(baseElement).toMatchSnapshot();
    })
    it("Profile Page without mock", () => {
        const {baseElement} = render(<Profile />);
        expect(baseElement).toMatchSnapshot();
    })
})