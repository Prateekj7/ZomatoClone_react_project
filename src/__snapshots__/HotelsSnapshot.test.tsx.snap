// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotels Sanpshot test Error Message 1`] = `
<body>
  <div>
    <div>
      <h5
        class="MuiTypography-root makeStyles-container-1 MuiTypography-h5"
        id="errorMessage"
      >
        test message
      </h5>
    </div>
  </div>
</body>
`;

exports[`Hotels Sanpshot test Loading Spinner 1`] = `
<body>
  <div>
    <div
      class="makeStyles-container-2"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`Hotels Sanpshot test Profile Page without mock 1`] = `
<body>
  <div>
    <div
      class="makeStyles-container-3"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        Welcome to your Profile!!!
      </h3>
      <h5
        class="MuiTypography-root MuiTypography-h5"
        id="displayName"
      />
      <h5
        class="MuiTypography-root MuiTypography-h5"
        id="email"
      />
      <h5
        class="MuiTypography-root MuiTypography-h5"
        id="uid"
      />
      <div
        class="makeStyles-container-4"
      >
        <button
          class="MuiButtonBase-root MuiFab-root makeStyles-button-5 MuiFab-sizeSmall MuiFab-extended MuiFab-primary"
          data-testid="primary"
          id="primary"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiFab-label"
          >
            Sign out
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiFab-root makeStyles-button-5 MuiFab-sizeSmall MuiFab-extended MuiFab-secondary"
          data-testid="secondary"
          id="secondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiFab-label"
          >
            Home
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-extended"
          data-testid="tertiary"
          id="tertiary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiFab-label"
          >
            Back
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
